### PR TITLE
Settings screen: improve discoverability of save button

### LINKED ISSUE #5353: Improve discoverability of “Save Changes” button on AMP settings screen
## Feature description

Given a viewport sized as following when looking at the AMP settings screen:

> image

Notice there is no Save button visible. This is even worse when a Reader theme is selected.

In the same way that there is a fixed bar at the bottom in the onboarding wizard, there may need to be something similar for the settings screen so that users can discover that they need to click “Save Changes”, without having to scroll all the way down the page.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by johnwatkins0**
## Summary

Fixes #5353

On the settings screen, this moves the "Save" button and the related success and error notices into a white bar fixed to the bottom of the screen. I have reused styles from the similar component in the onboarding wizard.

Result with no error:
Screen Shot 2020-09-28 at 8 34 14 AM

Result with error:
Screen Shot 2020-09-28 at 8 37 08 AM

## Checklist

- [x] My pull request is addressing an open issue (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by westonruter** [comment]
@johnwatkins0 As I chatted to you, there's a slight change in the height of the fixed bar when the notice is shown.

Notice Showing | Notice Hidden
-----------------|---------------
image | image

**Comment by johnwatkins0** [comment]
> there's a slight change in the height of the fixed bar when the notice is shown.
> 

Fixed in 327d2985c

File: assets/src/css/core-components.css [hunk 1]
```diff
@@ -154,3 +154,37 @@
...
+	display: flex;
+	justify-content: space-between;
+	margin: 0 auto;
+	max-width: 1320px;
+	padding: 15px;
+	width: 100%;
+
+	@media screen and (min-width: 783px) {
+		padding: 30px 90px;
+	}
```
Comment (westonruter) [inline assets/src/css/core-components.css:185]: Minor point: should files like this eventually get renamed to `*.sass`? This isn't valid CSS so my IDE complains:

image

**Review by westonruter — APPROVED** [review]
Works great for me! I will defer to @pierlon for confirming the JS changes.

File: assets/src/settings-page/settings-footer.js [hunk 2]
```diff
@@ -28,13 +101,27 @@ export function SettingsFooter() {
...
+				<Button isPrimary disabled={ disabled } isBusy={ savingOptions || downloadingTheme } type="submit">
+					{ savingOptions || downloadingTheme ? __( 'Saving', 'amp' ) : __( 'Save', 'amp' ) }
+					<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 64 64">
+						<path d="M43.16 10.18c-0.881-0.881-2.322-0.881-3.203 0s-0.881 2.322 0 3.203l16.335 16.335h-54.051c-1.281 0-2.242 1.041-2.242 2.242 0 1.281 0.961 2.322 2.242 2.322h54.051l-16.415 16.335c-0.881 0.881-0.881 2.322 0 3.203s2.322 0.881 3.203 0l20.259-20.259c0.881-0.881 0.881-2.322 0-3.203l-20.179-20.179z" />
+					</svg>
+				</Button>
+				{ error && <ErrorNotice errorMessage={ error.message || __( 'An error occurred. You might be offline or logged out.', 'amp' ) } /> }
+				{ ! error && ! hasOptionsChanges && didSaveOptions && ! downloadingTheme && (
+					<AMPNotice
+						className={ `amp-save-success-notice ${ savedNoticeClass }` }
```
Comment (pierlon) [inline assets/src/settings-page/settings-footer.js]: Shouldn't the notice be hidden once `savedNoticeClass` is set to `dismissed`?

File: assets/src/settings-page/settings-footer.js [hunk 3]
```diff
@@ -28,13 +101,27 @@ export function SettingsFooter() {
...
+		<section className="amp-settings-nav">
+			<div className="amp-settings-nav__inner">
+				<Button isPrimary disabled={ disabled } isBusy={ savingOptions || downloadingTheme } type="submit">
+					{ savingOptions || downloadingTheme ? __( 'Saving', 'amp' ) : __( 'Save', 'amp' ) }
+					<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 64 64">
+						<path d="M43.16 10.18c-0.881-0.881-2.322-0.881-3.203 0s-0.881 2.322 0 3.203l16.335 16.335h-54.051c-1.281 0-2.242 1.041-2.242 2.242 0 1.281 0.961 2.322 2.242 2.322h54.051l-16.415 16.335c-0.881 0.881-0.881 2.322 0 3.203s2.322 0.881 3.203 0l20.259-20.259c0.881-0.881 0.881-2.322 0-3.203l-20.179-20.179z" />
+					</svg>
+				</Button>
+				{ error && <ErrorNotice errorMessage={ error.message || __( 'An error occurred. You might be offline or logged out.', 'amp' ) } /> }
+				{ ! error && ! hasOptionsChanges && didSaveOptions && ! downloadingTheme && (
```
Comment (pierlon) [inline assets/src/settings-page/settings-footer.js:76]: Is it now the case that the notice is always to be shown unless this condition evaluates to `false`?

**Review by pierlon — CHANGES_REQUESTED** [review]

**Comment by pierlon** [comment]
Also, I noticed an error in one of the CSS files while reviewing (not related to these changes). Here's a patch with the fix:

```diff
--- assets/src/css/core-components.css	(revision 5b4b10be6bbf0729ec2eff5fce372d050466b367)
+++ assets/src/css/core-components.css	(date 1601691170646)
@@ -77,7 +77,7 @@
 
 .amp .components-button.is-primary:disabled.is-busy {
 	background-size: 100% 100%;
-	background-image: linear-gradient(-45deg, var(--amp-settings-color-brand) 28%, rgba(36, 89, 231, .8) 28%, rgb(36, 89, 231, .8) 72%, var(--amp-settings-color-brand) 72%);
+	background-image: linear-gradient(-45deg, var(--amp-settings-color-brand) 28%, rgba(36, 89, 231, .8) 28%, rgba(36, 89, 231, .8) 72%, var(--amp-settings-color-brand) 72%);
 	border-color: var(--color-gray-medium);
 }
 ```

File: assets/src/settings-page/settings-footer.js (hunk 3 above)
Comment (johnwatkins0) [reply assets/src/settings-page/settings-footer.js:76]: Yeah, it shows until the user makes changes again. So I just pushed 9d1ec07f5 to take this a step further and not have a custom class on the notice at all. That was to handle the fly in and automatic timeout, which aren't needed anymore.

File: assets/src/settings-page/settings-footer.js (hunk 2 above)
Comment (johnwatkins0) [reply assets/src/settings-page/settings-footer.js]: Removed the class entirely in 9d1ec07

**Comment by johnwatkins0** [comment]
> Also, I noticed an error in one of the CSS files while reviewing

Fixed in 468479031

File: tests/e2e/utils/onboarding-wizard-utils.js [hunk 4]
```diff
@@ -144,7 +144,7 @@ export function testTitle( { text, element = 'h1' } ) {
 }
 
 /**
- * Reset plugin configuration.
+ * Reset plugin configuration'.amp-settings-nav'
```
Comment (pierlon) [inline tests/e2e/utils/onboarding-wizard-utils.js]: I suppose this is a typo.

```suggestion
 * Reset plugin configuration.
```

**Review by pierlon — APPROVED** [review]

**Pull request merged**